datafusion/sql/src/planner.rs

```diff
@@ -2934,6 +2999,31 @@ fn parse_sql_number(n: &str) -> Result<Expr> {
         })
 }
 
+/// Wrap projection for a plan, if the join keys contains normal expression.
+fn wrap_projection_for_join_if_necessary(
+    join_keys: &[Expr],
+    input: LogicalPlan,
+) -> Result<LogicalPlan> {
+    let expr_join_keys = join_keys
+        .iter()
+        .flat_map(|expr| expr.try_into_col().is_err().then_some(expr))
+        .cloned()
+        .collect::<Vec<_>>();
+
+    let plan = if expr_join_keys.is_empty().not() {
```

Comment:
I found the use of `not()` confusing here -- I think it would be more consistent to use


```suggestion
    let plan = if !expr_join_keys.is_empty() {
```